### PR TITLE
Model update constructor

### PR DESCRIPTION
**Pull request by mfherbst**
Some of the improvements suggested in #675.

### CODE REVIEW

File: src/Model.jl [hunk 1]
```diff
@@ -203,6 +203,46 @@ function Model(system::AbstractSystem; kwargs...)
...
+          magnetic_moments=[],  # not used because symmetries explicitly given
+          terms=model.term_types,
+          model.temperature,
+          model.smearing,
+          model.εF,
+          model.spin_polarization,
+          model.symmetries,
+          # Can be safely disabled: this has been checked for model
+          disable_electrostatics_check=true,
+          kwargs...
```
Comment (antoine-levitt) [inline src/Model.jl:232]: This means the kwargs can't override the other settings (eg temperature etc). Maybe do a merge here?

File: src/Model.jl [hunk 2]
```diff
@@ -203,6 +203,46 @@ function Model(system::AbstractSystem; kwargs...)
     Model(parsed.lattice, parsed.atoms, parsed.positions; parsed.magnetic_moments, kwargs...)
 end
 
+"""
+    Model(model; [lattice, positions, atoms, kwargs...])
+    Model{T}(model; [lattice, positions, atoms, kwargs...])
+
+Construct an identical model to `model` with the option to change some of the contained
+parameters. If one of `lattice`, `positions`, `atoms` is missing, the respective value
+from `model` is used. This constructor is useful for changing the data type in the model
```
Comment (antoine-levitt) [inline src/Model.jl:212]: That sentence is kind of obvious no?

Comment (antoine-levitt) [reply src/Model.jl:212]: (I mean the "if one of ...")

File: src/fft.jl [hunk 3]
```diff
@@ -60,7 +60,7 @@ Perform a real valued iFFT; see [`ifft`](@ref).
 """
 function irfft(basis::PlaneWaveBasis{T}, f_fourier::AbstractArray; check=Val(true)) where {T}
     f_real = ifft(basis, f_fourier)
-    (check == Val(true)) && @assert norm(imag(f_real)) < sqrt(eps(T))
+    (check == Val(true)) && @assert norm(imag(f_real)) < max(1e-12, sqrt(eps(T)))
```
Comment (antoine-levitt) [inline src/fft.jl:63]: I hate these checks so much. Let's just kill them all...

File: src/postprocess/stresses.jl [hunk 4]
```diff
@@ -6,18 +6,7 @@ Compute the stresses (= 1/Vol dE/d(M*lattice), taken at M=I) of an obtained SCF
...
-                          magnetic_moments=[], # not used because symmetries explicitly given
-                          terms=model.term_types,
-                          model.temperature,
-                          model.smearing,
-                          model.εF,
-                          model.spin_polarization,
-                          model.symmetries,
-                          # Can be safely disabled: this has been checked for basis.model
-                          disable_electrostatics_check=true)
+        new_model = Model(basis.model; lattice)
```
Comment (antoine-levitt) [inline src/postprocess/stresses.jl:9]: nice

**Review by antoine-levitt — COMMENTED** [review]
Cool!

File: src/Model.jl (hunk 1 above)
Comment (mfherbst) [reply src/Model.jl:232]: Interestingly not. The `kwargs` take preference.

File: src/fft.jl (hunk 3 above)
Comment (mfherbst) [reply src/fft.jl:63]: Not sure. They do help during development for sure.

File: src/Model.jl (hunk 1 above)
Comment (antoine-levitt) [reply src/Model.jl:231]: Really?
```
julia> sum([]; dims=1, dims=2)
ERROR: syntax: keyword argument "dims" repeated in call to "sum" around REPL[1]:1
```

Comment (mfherbst) [reply src/Model.jl:231]: Yeah but this does not use a `kwargs...` construct. That makes a difference.

Comment (antoine-levitt) [reply src/Model.jl:231]:
```
julia> kwargs = (; dims=1)
(dims = 1,)

julia> sum(; kwargs..., dims=2)
ERROR: MethodError: no method matching sum(; dims=2)
```
Woah this is wild. Is this documented anywhere?

**Pull request merged**